### ISSUE TITLE
Removed relations do not appear in the log

### ISSUE THREAD

**Issue by msimkunas** [jacobdekeizer/oc-logbook#3]
Setting `showLogRelations` to `true` enables the logging of changes to related models. This does not, however, track the removal of related models. Is this the intended behavior?

**Comment by jacobdekeizer** [comment]
Hi @msimkunas, does every related model use the LogChanges trait and has this property set to false `protected $deleteLogbookAfterDelete = false;`. This should log a new entry when the model is deleted.

**Comment by msimkunas** [comment]
Thanks, that was exactly what I was missing.

This only works if the related models are soft deleted though. I had to add the `SoftDelete` trait to the related model AND the `withTrashed` scope to the relation definition of the parent model to make this work.

For the time being, I've simply defined an additional relation on my parent model that I only use for audit logs. This results in a parent model `Foo` with relations `bar` and `bar_with_trashed`. I resorted to this workaround because I didn't want to return trashed models in my `bar` relation.

It's a viable workaround but I feel that the plugin could be improved to take soft deleted models into account. The following lines of code seem to be the culprit: https://github.com/jacobdekeizer/oc-logbook/blob/master/traits/LogChanges.php#L219-L224:
```
$relatedModels = $this->$relation;

// no related items found
if ($relatedModels === null ) {
    continue;
}
```

Changing L219 to `$relatedModels = $this->$relation()->withTrashed()->get();` seems to solve the problem. Since this is a breaking change, an additional option could be added to the form widget which would enable this behavior (e.g. `showSoftDeletedRelations`);

This change would also require additional handling of the `undo` action when triggering it for soft deleted relations. I haven't tested this extensively but the undo action doesn't seem to do anything when the related model is soft deleted. It should either restore the related model or be disabled altogether.

**Comment by jacobdekeizer** [comment]
@msimkunas thanks for pointing this out, I will look into this.

**Comment by jacobdekeizer** [comment]
@msimkunas I fixed the problem and tagged a new version 2.0 where I refactored a lot of code. The formwidget has now support for soft deleted models with the `showSoftDeleteRelations` as described in the readme. Let me know if it works for you :)

**Comment by msimkunas** [comment]
I've just tested the new functionality and can confirm that it works perfectly well and my issue is solved. Thanks!
